AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -723,11 +724,27 @@ private static void handleApiResponse(JSONObject response, List<Map<String, Obje
       for (int i = requests.size() - 1; i >= 0; i--) {
         Map<String, Object> currentRequest = requests.get(i);
         if (Constants.Methods.START.equals(currentRequest.get(Constants.Params.ACTION))) {
-          if (i < responseCount) {
-            lastStartResponse = Request.getResponseAt(response, i);
+          if (currentRequest.containsKey(Constants.Params.REQ_ID)) {
+              for (int j = Request.numResponses(response) - 1; j >= 0; j--) {
```

Comment:
out of curiosity, why are you looping from `n-1` to `0` instead of the other way?

Reply:
gracefully degrade - If we end up having multiple requests with the same ID, we pick the latest request with that id.

Reply:
Could this loop be 10k long? Do we have any speculation on performance?